api/types/network/network.go

```diff
@@ -100,3 +100,13 @@ func (es *EndpointSettings) Copy() *EndpointSettings {
 type NetworkingConfig struct {
 	EndpointsConfig map[string]*EndpointSettings // Endpoint configs for each connecting network
 }
+
+// NetworkConfigReference specifies the source which provides a network's configuration
+type NetworkConfigReference struct {
+	Source NetworkReference
```

Comment:
Can you rename this field to `Network` and make it a pointer? The idea is that we may support other sources in the future, in which case there would be other fields that provided alternative sources.